source/server/filter_chain_manager_impl.h

```diff
@@ -365,6 +370,9 @@ class FilterChainManagerImpl : public Network::FilterChainManager,
   // detect the filter chains in the intersection of existing listener and new listener.
   FcContextMap fc_contexts_;
 
+  // Mapping from filter chain name to filter chain.
+  absl::flat_hash_map<std::string, Network::DrainableFilterChainSharedPtr> filter_chains_by_name_;
```

Comment:
Can you move this field close to `matcher_`?

Basically, one of below is activated, never both
1. destination_ports_map_, or
2. matcher_ + filter_chains_by_name_

Reply:
Done.